receiver/builder.go

```diff
@@ -0,0 +1,97 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package receiver // import "go.opentelemetry.io/collector/receiver"
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+)
+
+// Builder receiver is a helper struct that given a set of Configs and Factories helps with creating receivers.
+type Builder struct {
```

Comment:
Curious why aren't we doing it for the `Builder`?

Reply:
Because it's a struct, not an interface. So we don't have a choice, but to make its methods available.

Reply:
I see. I don't really like that we are doing it to only partially hide the new experimental API while the other parts are being added as is. Seems like a questionable effort to me. I would leave it as is and add profiling parts where needed, including the interface. I doubt they are going to be changed in any way. However, I'm not going to argue about it if this approach is something that was decided already

Reply:
> I see. I don't really like that we are doing it to only partially hide the new experimental API while the other parts are being added as is

We don't have to and are not going to partially hide the API. Adding a method to get the config from a receiver would be sufficient to add support for profiles without exposing experimental methods. Getting rid of the builders (making them an internal impl detail in `service`) is also an option.

Reply:
See #10621 and https://github.com/open-telemetry/opentelemetry-collector/pull/10375#pullrequestreview-2144929463